backends/core/models.py

```diff
@@ -249,9 +275,22 @@ def destroy(self):
       Param.destroy(*param_ids)
     self.delete()
 
-  def get_ready(self):
-    if self.status not in ['idle', 'succeeded', 'failed']:
+  def get_status(self):
+    key = '%s%s%s' % (self._get_pipeline_prefix(), self._get_job_prefix(), CACHE_KEY_STATUS)
+    return memcache_client.get(key, default_value=self.status)
+
+  def prepare_for_start(self):
+    """
+      Check the current status of the job and update for 'waiting'
+    """
+    status = self.get_status()
+    if status not in ['idle', 'succeeded', 'failed']:
       return False
+    key = '%s%s%s' % (self._get_pipeline_prefix(), self._get_job_prefix(), CACHE_KEY_STATUS)
```

Comment:
The method `add` used in `get_or_create_cache` does not have a prefix argument, so it's just moving the concatenation in this method:
`key = '%s%s' % (prefix, key)`
What do you think?
As for the preference to split into pipeline prefix and job prefix, it is because there are fields like `failed_jobs` that only need the pipeline prefix and fields like `status` that need both pipeline and job prefix. I suppose I can change the `_get_job_prefix()` to 
`return '%s_%s_' % (str(self.pipeline_id), str(self.id))` so that I would need 2 method calls for this prefix.

Reply:
done